pkg/sql/lease.go

```diff
@@ -581,26 +593,35 @@ func (t *tableState) acquire(
 	t.mu.Lock()
 	defer t.mu.Unlock()
 
+	// When a lease is about to expire but has not yet (determined by
+	// leaseHasEnoughLifeLeft), then ensure an asynchronous routine to renew the
+	// lease is started.
+	if s := t.mu.active.findNewest(); s != nil && !s.hasExpired(timestamp) && !s.leaseHasEnoughLifeLeft() && t.mu.acquiring == nil {
```

Comment:
I don't believe you need hasExpired() here. If it's true you'll acquire the lease here and wait in acquireWait() below